makefile: add a new target to run 'golangci-lint run --fix'

aspsk opened the pull request:
There is a make target to execute the 'golangci-lint run' command, but there is no simple way to execute it with a '--fix' argument. Add a new target to do this.

If more arguments need to be added to the golangci-lint command, then one can do this by setting the newly added GOLANGCI_LINT_ARGS variable.

aspsk committed:
makefile: add a new target to run 'golangci-lint run --fix'

There is a make target to execute the 'golangci-lint run' command, but there is
no simple way to execute it with a '--fix' argument. Add a new target to do
this.

If more arguments need to be added to the golangci-lint command, then one can
do this by setting the newly added GOLANGCI_LINT_ARGS variable.

Signed-off-by: Anton Protopopov <aspsk@isovalent.com>

joestringer commented on Makefile:
```diff
@@ -479,14 +479,19 @@ govet: ## Run govet on Go source files in the repository.
+
 lint: golangci-lint ## Run golangci-lint and bpf-mock linters.
 
+lint-fix: golangci-lint-fix ## Run golangci-lint and bpf-mock linters.
```
Does bpf-mock still exist?

aspsk replied:
Thanks, I've removed the comments mentioning it

aspsk committed:
makefile: cleanup comments

The commit 7303c02bc2 ("bpf-mock makefile: Remove removed mock target") removed
the bpf-mock-lint target, but there left some comments about running it. Remove
the mentions of bpf-mock.

Signed-off-by: Anton Protopopov <aspsk@isovalent.com>
